src/core.js

```diff
@@ -90,19 +90,12 @@ class Container {
   }
 
   teardown() {
-    let effects = this[effectsSymbol];
-    if(effects) {
-      for(let effect of effects) {
-        effect.teardown();
-      }
-    }
-
-    let contexts = this[contextSymbol];
-    if(contexts) {
-      for(let context of contexts) {
-        context.unsubscribe();
```

Comment:
Can you explain why the effects/context teardown isn't happening now?

Reply:
Instead of making the teardown step specific to useEffect and useContext, I made a change to make it applicable to any hook that implemented a teardown() method.  I could see a similar approach being used for the effects step too...